rskj-core/src/main/java/co/rsk/rpc/modules/eth/subscribe/LogsNotification.java

```diff
@@ -34,64 +35,97 @@
+    private final int transactionIndex;
     private final boolean removed;
 
+    private String lazyLogIndex;
+    private String lazyBlockNumber;
+    private String lazyBlockHash;
+    private String lazyTransactionHash;
+    private String lazyTransactionIndex;
+    private String lazyAddress;
+    private String lazyData;
+    private List<String> lazyTopics;
+
     public LogsNotification(LogInfo logInfo, Block b, int txIndex, Transaction tx, int logIdx, boolean r) {
-        logIndex = toQuantityJsonHex(logIdx);
-        blockNumber = toQuantityJsonHex(b.getNumber());
-        blockHash = b.getHashJsonString();
-        removed = r;
-        transactionIndex = toQuantityJsonHex(txIndex);
-        transactionHash = tx.getHash().toJsonString();
-        address = toJsonHex(logInfo.getAddress());
-        data = toJsonHex(logInfo.getData());
-        topics = logInfo.getTopics().stream()
-                .map(t -> toJsonHex(t.getData()))
-                .collect(Collectors.toList());
+        this.logInfo = logInfo;
+        this.block = b;
+        this.transaction = tx;
+        this.logInfoIndex = logIdx;
+        this.removed = r;
+        this.transactionIndex = txIndex;
     }
 
     public String getLogIndex() {
-        return logIndex;
+        if (lazyLogIndex == null) {
+            lazyLogIndex = toQuantityJsonHex(logInfoIndex);
+        }
+        return lazyLogIndex;
     }
 
     public String getBlockNumber() {
-        return blockNumber;
+        if (lazyBlockNumber == null) {
+            lazyBlockNumber = toQuantityJsonHex(block.getNumber());
+        }
+        return lazyBlockNumber;
     }
 
     public String getBlockHash() {
-        return blockHash;
+        if (lazyBlockHash == null) {
+            lazyBlockHash = block.getHashJsonString();
+        }
+        return lazyBlockHash;
     }
 
     public String getTransactionHash() {
-        return transactionHash;
+        if (lazyTransactionHash == null) {
+            lazyTransactionHash = transaction.getHash().toJsonString();
+        }
+        return lazyTransactionHash;
     }
 
     public String getTransactionIndex() {
-        return transactionIndex;
+        if (lazyTransactionIndex == null) {
+            lazyTransactionIndex = toQuantityJsonHex(transactionIndex);
+        }
+        return lazyTransactionIndex;
     }
 
     public String getAddress() {
-        return address;
+        if (lazyAddress == null) {
+            lazyAddress = toJsonHex(logInfo.getAddress());
+        }
+        return lazyAddress;
     }
 
     public String getData() {
-        return data;
+        if (lazyData == null) {
+            lazyData = toJsonHex(logInfo.getData());
+        }
+        return lazyData;
     }
 
     public List<String> getTopics() {
-        return Collections.unmodifiableList(topics);
+        if (lazyTopics == null) {
+            lazyTopics = logInfo.getTopics().stream()
+                    .map(t -> toJsonHex(t.getData()))
```

Comment:
How about:

```suggestion
                    .map(Topic::toJsonString)
```

Reply:
`Topic::toJsonString` doesn't have the `0x00` expansion for _null_ values :(

Reply:
Then isn't _that_ wrong?